### PR TITLE
Tables & Chairs (sorry no chairs actually)

### PR DESCRIPTION
**Pull request by ethteck**
Sorted out the battle item/move tables, converted bss c files to asm, misc decomp, etc

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 21 functions (+0.30%) on `us`.

✅ This PR fixes 16 warnings!

**Comment by HailSanta** [comment]
where are the chairs

**Comment by ethteck** [comment]
there aren't any im sorry

File: src/8a860_len_3f30.c [hunk 1]
```diff
@@ -88,7 +89,68 @@ void func_800F4C6C(PopupMessage* popup, s32 x, s32 y);
...
+    if (gPopupMenu->popupType == POPUP_MENU_TRADE_FOR_BADGE) {
+        set_window_update(WINDOW_ID_16, (s32) basic_hidden_window_update);
+        set_window_update(WINDOW_ID_20, (s32) basic_hidden_window_update);
+    }
+    if (gPopupMenu->popupType == POPUP_MENU_UPGRADE_PARTNER) {
+        set_window_update(WINDOW_ID_18, (s32) basic_hidden_window_update);
+    }
+    if (gPopupMenu->popupType == POPUP_MENU_SELL_ITEM) {
+        set_window_update(WINDOW_ID_16, (s32) basic_hidden_window_update);
+        set_window_update(WINDOW_ID_20, (s32) basic_hidden_window_update);
```
Comment (nanaian) [inline src/8a860_len_3f30.c:149]: We really need to retype this func to allow func ptrs as well as ints

**Review by nanaian — APPROVED** [review]

**Comment by nanaian** [comment]
add chairs please

File: src/8a860_len_3f30.c (hunk 1 above)
Comment (ethteck) [reply src/8a860_len_3f30.c:149]: It's difficult, because doing so has messed with codegen unfortunately :/ but yeah we should revisit this for sure

**Pull request merged**